p2p/pkg/rpc/validator/service_test.go

```diff
@@ -286,47 +292,58 @@ func TestProcessEpoch(t *testing.T) {
 	slotDuration := 12 * time.Second
 	svc := validatorapi.NewService(ts.URL, mockValidatorRouter, logger, optsGetter, mockNotifier, notifyOffset)
 
-	svc.SetGenesisTime(time.Now().Add(100*time.Millisecond - slotDuration))
+	svc.SetGenesisTime(time.Now().Add(1*time.Second + 500*time.Millisecond - slotDuration))
 
 	svc.SetProcessEpoch(ctx, 10, time.Now().Unix())
 
-	select {
-	case n := <-mockNotifier.NotifyCh:
-		if n == nil {
-			t.Fatal("expected notification, got nil")
-		}
-		if n.Topic() == notifications.TopicValidatorOptedIn {
-			slotVal, ok := n.Value()["slot"]
-			if !ok {
-				t.Fatal("expected slot in notification value")
-			}
-			if slotVal != uint64(1) {
-				t.Errorf("expected slot 1, got %v", slotVal)
-			}
-		} else if n.Topic() == notifications.TopicEpochValidatorsOptedIn {
-			slotsVal, ok := n.Value()["slots"]
-			if !ok {
-				t.Fatal("expected slots in notification value")
-			}
-			slotsSlice, ok := slotsVal.([]any)
-			if !ok {
-				t.Fatalf("expected slots to be a slice of maps, got %T: %v", slotsVal, slotsVal)
-			}
-			if len(slotsSlice) != 1 {
-				t.Fatalf("expected 1 slot, got %d", len(slotsSlice))
+	numValidatorOptedInNotifs := 0
+	numEpochValidatorsOptedInNotifs := 0
+
+	timeout := time.After(1*time.Second + slotDuration)
+
+	// expect 2 validator opted in and 1 epoch notif
+	for numValidatorOptedInNotifs < 2 || numEpochValidatorsOptedInNotifs < 1 {
```

Comment:
Previously we didn't have a for-loop here, so only the first TopicEpochValidatorsOptedIn notification was checked, and the test passed without any TopicValidatorOptedIn notifications